spec/models/push_spec.rb

```diff
@@ -1,33 +1,47 @@
 require 'spec_helper'
 
 describe 'Push' do
-  def payload
-    @payload ||= Github::Api::PushHookPayload.new(load_json_fixture('github_push_payload'))
+  before(:all) do
+    AncestorRef.create!([
+                          { service_name: 'web1', ref: 'production' },
+                          { service_name: 'rs1',  ref: 'a_whole_new_ref' }
+                        ])
   end
 
+  let(:ancestor_refs) { AncestorRef.all }
+  let(:payload) { Github::Api::PushHookPayload.new(load_json_fixture('github_push_payload')) }
+
   it 'can create be constructed from github data' do
-    push = Push.create_from_github_data!(payload)
-    expect(push.status).to eq(Github::Api::Status::STATE_PENDING.to_s)
-    expect(push.head_commit).not_to be_nil
-    expect(push.branch).not_to be_nil
-    expect(push.commits.count).to eq(1)
-    expect(push.jira_issues.count).to eq(0)
-    expect(push.created_at).not_to be_nil
-    expect(push.updated_at).not_to be_nil
-    expect(push.email_sent).to eq(false)
+    pushes = Push.create_from_github_data!(payload)
```

Comment:
Should this test also confirm the services for each push?

Reply:
updated 👍